Newspack/Newspack/Stores/MediaImporter.swift

```diff
@@ -0,0 +1,315 @@
+
+        // Remove Orientation data
+        if var tiffProps = properties[Constants.tiffDictionaryKey] as? [String: Any] {
+            tiffProps.removeValue(forKey: Constants.tiffOrientationKey)
+            properties[Constants.tiffDictionaryKey] = tiffProps
+        }
+
+        if var iptcProps = properties[Constants.iptcDictionaryKey] as? [String: Any] {
+            iptcProps.removeValue(forKey: Constants.iptcOrientationKey)
+            properties[Constants.iptcDictionaryKey] = iptcProps
+        }
+
+        return properties
+    }
+
+    /// Returns a new, prepared, instance of the supplied CIImage instance.
+    ///
+    /// - Parameter image: The image to sanitize.
+    /// - Returns: A CIImage instance.
+    ///
+    func prepareImage(image: CIImage) -> CIImage {
+        let properties = sanitizeImageMetaData(metaData: image.properties)
+        return image.settingProperties(properties)
+    }
+
+    /// Copy the file backing a PHAsset to a local directory in preparation for uploading.
+    ///
+    /// - Parameter asset: A PHAsset instance.  An image is expected.
+    /// - Parameter contentEditingInput: A PHContentEditingInput instance
+    /// - Returns: The file URL for the copied asset or nil.
+    ///
+    func copyAssetToFile(asset: PHAsset, contentEditingInput: PHContentEditingInput) throws -> URL? {
+        guard
+            let originalFileURL = contentEditingInput.fullSizeImageURL,
+            let originalImage = CIImage(contentsOf: originalFileURL),
+            let uti = contentEditingInput.uniformTypeIdentifier
+        else {
+            throw MediaImporterError.unableToWriteFile
+        }
+
+        let image = prepareImage(image: originalImage)
+        let fileURL = importURLForOriginalURL(originalURL: originalFileURL)
+
+        try writeImage(image: image, withUTI: uti, toFile: fileURL)
+
+        return fileURL
+    }
+
+    /// Find a usable import URL for the original URL. If a file already exists
+    /// at a candidate URL, a numeric suffix is added. The suffix will be
+    /// incremented until an available filename is found.
+    ///
+    /// - Parameter originalURL: The original file URL.
+    /// - Returns: An available URL for an imported asset.
+    ///
+    func importURLForOriginalURL(originalURL: URL) -> URL {
+        let fileManager = FileManager()
+
+        var candidateURL = directoryPath.appendingPathComponent(originalURL.lastPathComponent)
+        guard fileManager.fileExists(atPath: candidateURL.path) else {
+            return candidateURL
+        }
+
+        let filename = originalURL.lastPathComponent.components(separatedBy: ".").first!
+        var counter = 1
+        repeat {
+            let path = "\(filename)-\(counter).\(originalURL.pathExtension)"
+            candidateURL = directoryPath.appendingPathComponent(path)
+            counter = counter + 1
+        } while fileManager.fileExists(atPath: candidateURL.path)
```

Comment:
Hmm... I don't think so, at least not the enumerator since I'd still need to do the work of generating and checking for an available filename.  Seems like there should be something automated in the api for this but I haven't found it. 🤷

Reply:
Ahhhh gotcha!!! Makes 💯  sense!!